clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2138,6 +2138,8 @@ static void handleConstructorAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   if (AL.getNumArgs() &&
       !S.checkUInt32Argument(AL, AL.getArgAsExpr(0), priority))
     return;
+  S.Diag(D->getLocation(), diag::warn_global_constructor)
```

Comment:
Why is this firing always?  I thought this is only supposed to fire on Global and ThreadLocal variables according to the GCC patch summary.  But this will fire any time there is a `ConstructorAttr` or `DestructorAttr` somewhere.

Reply:
How I see it is:

From GCC's docs: "The constructor attribute causes the function to be called automatically before execution enters main ()."

And `-Wglobal-constructors` is meant to tell you "this has a constructor which will fire before main" which matters for global and thread local variables, so the `constructor` attribute is sort of morally equivalent.

Reply:
Ah!  I missed that about the constructor attribute.  Thanks!  Makes sense to me.